src/plugins/embeddable/public/add_panel_flyout/add_panel_flyout.tsx

```diff
@@ -97,40 +82,28 @@ export const AddPanelFlyout = ({
       name: string,
       savedObject: SavedObjectCommon
     ) => {
-      const factoryForSavedObjectType = factoriesBySavedObjectType[type];
-      if (!factoryForSavedObjectType) {
-        throw new EmbeddableFactoryNotFoundError(type);
+      if (factoriesBySavedObjectType[type]) {
+        const factory = factoriesBySavedObjectType[type];
+        const { method } = factory;
+        method(container, savedObject);
+        return;
       }
 
-      let embeddableId: string;
-
-      if (savedObjectToPanel[type]) {
-        // this panel type has a custom method for converting saved objects to panels
-        const panel = savedObjectToPanel[type](savedObject);
-
-        const { id: _embeddableId } = await container.addNewEmbeddable(
-          factoryForSavedObjectType.type,
-          panel,
-          savedObject.attributes
-        );
-
-        embeddableId = _embeddableId;
-      } else {
-        const { id: _embeddableId } = await container.addNewEmbeddable<SavedObjectEmbeddableInput>(
-          factoryForSavedObjectType.type,
-          { savedObjectId: id },
-          savedObject.attributes
-        );
-
-        embeddableId = _embeddableId;
+      const legacyFactoryForSavedObjectType = legacyFactoriesBySavedObjectType[type];
+      if (!legacyFactoryForSavedObjectType) {
+        throw new EmbeddableFactoryNotFoundError(type);
       }
 
-      onAddPanel?.(embeddableId);
+      const initialState = savedObjectToPanel[type](savedObject) ?? { savedObjectId: id };
+
+      container.addNewPanel({
+        panelType: legacyFactoryForSavedObjectType.type,
+        initialState,
+      });
 
-      showSuccessToast(name);
-      runAddTelemetry(container.type, factoryForSavedObjectType, savedObject);
```

Comment:
How come you removed telemetry?

Reply:
I replaced `container.addNewEmbeddable` with `container.addNewPanel` because `container.addNewPanel` [reports telemetry](https://github.com/elastic/kibana/blob/main/src/plugins/dashboard/public/dashboard_container/embeddable/dashboard_container.tsx#L438-L440). 

However, now I notice the event name is different. Previously `runAddTelemetry` uses the event name `{panelType}:add`, e.g. `maps:add`. But `addNewPanel` uses the event name `{panelType}`, e.g. `maps`. 

Perhaps `addNewPanel` should remain consistent by using `{panelType}:add`?

Reply:
A gave this more thought and I will restore the `runAddTelemetry` function here. The `container.addNewPanel` should be considered a separate event name as it can be triggered by both the "Add from library" and "Create panel" buttons. The "Create panel" button also publishes telemetry under the event name `{panelType}:create`. 

These are the telemetry events that will be published in both workflows:

## "Add from library"
1. When a user clicks on a library item in the flyout `{panelType}:add` is published.
2. `{panelType}` is also published by `container.addNewPanel`.

## "Create panel" dropdown
1. When a user clicks on a panel type in the `{panelType}:create` is published.
2. If the user cancels the creation process, no other events are published.
3. If the user completes the creation process, `{panelType}` is published by `container.addNewPanel`.

Reply:
Fixed in b140b78d5292d4082ca7ee9ae2e4214ddf93c85f